app/src/main/java/org/onionshare/android/ShareManager.kt

```diff
@@ -59,8 +59,12 @@ class ShareManager @Inject constructor(
         if (f is FilesState.Added && t is TorState.Stopped && w is WebServerState.Stopped && !w.downloadComplete) {
             if (f.files.isEmpty()) emit(ShareUiState.NoFiles)
             else emit(ShareUiState.FilesAdded(f.files))
-        } // handle error while adding files
-        else if (f is FilesState.Error) {
+        } // handle error while adding files while Tor is still starting or started
+        else if (f is FilesState.Error && (t is TorState.Starting || t is TorState.Started)) {
+            stopSharing()
```

Comment:
If we don't stop, Tor would keep running and not match the UI.

Reply:
The error will be shown as soon as Tor has stopped as we don't reset the error state before.